frame/staking/src/benchmarking.rs

```diff
@@ -285,13 +361,19 @@ benchmarks! {
 			<ErasTotalStake<T>>::insert(i, BalanceOf::<T>::one());
 			ErasStartSessionIndex::insert(i, i);
 		}
-	}: _(RawOrigin::Root, EraIndex::zero())
+	}: _(RawOrigin::Root, EraIndex::zero(), u32::max_value())
+	verify {
+		assert_eq!(HistoryDepth::get(), 0);
+	}
 
 	reap_stash {
 		let u in 1 .. 1000;
 		let (stash, controller) = create_stash_controller::<T>(u)?;
 		T::Currency::make_free_balance_be(&stash, 0.into());
-	}: _(RawOrigin::Signed(controller), stash)
+	}: _(RawOrigin::Signed(controller), stash.clone())
```

Comment:
As I see, this doesnt include all the cost of `slashing::clear_stash_metadata::<T>(stash);`

actually there should be `db writes += number of span`

Reply:
I dont think this applies right?

You can only call `reap_stash` when the user's account `total_balance` is zero.

```
		#[weight = 70 * WEIGHT_PER_MICROS + T::DbWeight::get().reads_writes(4, 7)]
		fn reap_stash(_origin, stash: T::AccountId) {
			ensure!(T::Currency::total_balance(&stash).is_zero(), Error::<T>::FundedTarget);
			Self::kill_stash(&stash)?;
			T::Currency::remove_lock(STAKING_ID, &stash);
		}

Reply:
not familiar myself but I think it applies: in this test: https://github.com/paritytech/substrate/blob/fe67a0c41884bda918b55b658a041588a5014a3e/frame/staking/src/tests.rs#L2273

two element of `SpanSlash` are removed

Reply:
fixed! good catch